index.js

```diff
@@ -88,11 +95,24 @@ function retryRequest(requestOpts, opts, callback) {
       delayStream = through({ objectMode: opts.objectMode });
       requestStream = opts.request(requestOpts);
 
-      requestStream
-        .on('error', onResponse)
-        .on('response', onResponse.bind(null, null))
-        .on('complete', retryStream.emit.bind(retryStream, 'complete'))
-        .pipe(delayStream);
+      var events = {
+        error: onResponse,
+        response: onResponse.bind(null, null),
+        complete: retryStream.emit.bind(retryStream, 'complete')
+      };
+
+      for (var eventName in events) {
+        requestStream.once(eventName, events[eventName]);
+      }
+
+      removeAllListeners = function() {
+        for (var eventName in events) {
+          requestStream.on(eventName, function() {});
```

Comment:
I don't fully understand this one-- without this, we get an uncaught error event. But the only reason we are unregistering our these events is because our error handler was already fired. So, I can't figure out why the error would fire again.